Public review draft tech arch vision

WesIngwersen opened the pull request:
This PR serves as a public review of the Technical Architecture Vision paper. View [the draft vision in a legible format](https://github.com/cornerstone-data/papers/blob/public-review-draft-tech-arch-vision/architecture_vision/Cornerstone_Architecture_Vision.md). 

To comment on specific line:
Look under Files changed and the Cornerstone_Architecture_Vision.md and then click the `+` symbol on the line of your choice to add a comment. 

For general comments, just add them below.

cclin130 committed:
architecture vision md

cclin130 committed:
cleanup

cclin130 committed:
one last touchup

cclin130 committed:
fix arrow in header

cclin130 committed:
remove google doc link

WesIngwersen committed:
Add link to Cornerstone Technical Architecture Vision

Added a link to the Cornerstone Technical Architecture Vision document.

WesIngwersen committed:
Add status of Cornerstone Technical Architecture Vision paper

WesIngwersen committed:
resolves #`

cclin130 committed:
address comments

bl-young committed:
add footnote

WesIngwersen committed:
Use polyrepo spelling to match monorepo; add monorepo reference; change emphasis words to be consistent with emphasizing key words and not ideas. Fix heading above fig 4 to read more like a heading. Make vision unlocks structure more consistent by updating the bullet names and structure.

WesIngwersen committed:
consistency - use bold for key terms and italics for emphasis except for when used as subheaders

WesIngwersen committed:
Add USEEIO references. Link to cornerstone-controlled sites except for papers.

cclin130 committed:
Merge pull request #1 from cornerstone-data/tech-arch-vision-draft

Tech arch vision draft - internal review

MoLi7 committed:
update file structure figure

staadecker commented:
Hi Wes,

Not sure if this is the right place to provide feedback however I thought I'd ask what Python libraries Cornerstone was planning to use in its data pipelines? Specifically, I know pandas is often used, however, I would strongly recommend [Polars](https://pola.rs/) as an alternative library for manipulating DataFrames.

Polars is much faster than pandas since its backend is written in Rust and it multi-threads all computations by splitting DataFrames into chunks that are each processed by a different core. (It also has a query engine that can simplify chained queries). Most importantly, Polars' syntax is extremely legible and powerful (no pandas indexes to mess with or in-place operations). I've come to love Polars over pandas and simply thought I'd let the team know :)

staadecker commented:
Otherwise, I think Python, monorepo, and ETL pipeline are excellent calls! I also love the `.yaml` based approach that I'm seeing in the bedrock repo.

catherinebirney committed:
upsate file structure

catherinebirney committed:
drop old file structure image

catherinebirney committed:
add updated file structure image

WesIngwersen commented:
> Hi Wes,
> 
> Not sure if this is the right place to provide feedback however I thought I'd ask what Python libraries Cornerstone was planning to use in its data pipelines? Specifically, I know pandas is often used, however, I would strongly recommend [Polars](https://pola.rs/) as an alternative library for manipulating DataFrames.
> 
> Polars is much faster than pandas since its backend is written in Rust and it multi-threads all computations by splitting DataFrames into chunks that are each processed by a different core. (It also has a query engine that can simplify chained queries). Most importantly, Polars' syntax is extremely legible and powerful (no pandas indexes to mess with or in-place operations). I've come to love Polars over pandas and simply thought I'd let the team know :)

Thanks @staadecker for this comment. We don't get into specifics of supporting Python packages in this paper but we are looking into drawing on this to speed up handling of large matrices.  Great suggestion.

WesIngwersen committed:
Minor wording/timing change in intro

WesIngwersen committed:
drop duplicate Rho

WesIngwersen committed:
minor fix of matrix list language

WesIngwersen committed:
add missing close paren

WesIngwersen committed:
minor language and formatting fixes

- remove unneeded contractions
- fix final bulleted section formatting

WesIngwersen committed:
drop matrix requirements which don't fit in scope

WesIngwersen committed:
minor grammar fixes

- remove contractions
- use "document" instead of doc
other sentence structure changes
